src/services/telemetryService.ts

```diff
@@ -71,12 +71,9 @@ export class TelemetryService extends Disposable implements ITelemetryService {
 	private getSegmentAnalyticsClient(gitpodHost: string, segmentKey: string): Analytics | undefined {
 		const serviceUrl = new URL(gitpodHost);
 		if (this.analitycsClients.has(serviceUrl.host)) {
-			return this.analitycsClients.get(serviceUrl.host)!;
+			return this.analitycsClients.get(serviceUrl.host);
 		}
 		const client = createSegmentAnalyticsClient({ writeKey: segmentKey }, gitpodHost, this.logService);
-		if (!client) {
```

Comment:
Fix nit problem that it will always try to `createSegmentAnalyticsClient` if map is not set. Then for dedicated, it will log error https://github.com/gitpod-io/gitpod-vscode-desktop/blob/18f8a32f3b2d878acec979a03add53015121b1d6/src/common/telemetry.ts#L56